File: 30 Days of Code/Day 18 Queues and Stacks/palindrome.py
class Solution:
	def __init__(self):
		self.q = list()
		self.s = list()

	def enqueueCharacter(self, c):
		self.q.append(c)

	def dequeueCharacter(self):
		return self.q.pop(0)

	def pushCharacter(self, c):
		self.s.append(c)

	def popCharacter(self):
		return self.s.pop()

str = input()
obj = Solution()

l = len(str)

for s in str:
	obj.pushCharacter(s)
	obj.enqueueCharacter(s)

isPalindrome = True

for i in range(l//2):
	if obj.popCharacter() != obj.dequeueCharacter():
		isPalindrome = False
		break

print("The word, " + str + ", is a palindrome." \
	  if isPalindrome \
	  else \
	  "The word, " + str + ", is not a palindrome.")

File: 30 Days of Code/Day 18 Queues and Stacks/Palindrome.java
import java.util.Scanner;
import java.util.LinkedList;

/* Using LinkedList to implement queue and stack*/
public class Palindrome {
	private LinkedList<Character> q = new LinkedList<Character>();
	private LinkedList<Character> s = new LinkedList<Character>();

	void enqueueCharacter(char c) {
		q.add(c);
	}

	char dequeueCharacter() {
		return (char)q.removeFirst();
	}

	void pushCharacter(char c) {
		q.add(c);
	}

	char popCharacter() {
		return (char)q.removeLast();
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		String input = scan.nextLine();
		scan.close();

		char[] s = input.toCharArray();

		Palindrome p = new Palindrome();

		for (char c : s) {
			p.pushCharacter(c);
			p.enqueueCharacter(c);
		}

		boolean isPalindrome = true;
		for (int i = 0; i < s.length/2; i++) {
			if (p.popCharacter() != p.dequeueCharacter()) {
				isPalindrome = false;
				break;
			}
		}

		System.out.println("The word, " + input + ", is "
						   + ((!isPalindrome) ? "not a palindrome." : "a palindrome."));
	}
}